### PR TITLE
274 getgasprice

### LINKED ISSUE #274: GetGasPrice
We create a best practice code that :
    It takes in a request req and a response res. The request contains the parameter network, which represents the blockchain network for which you want to fetch the gas price.

    The network parameter from the request is converted to uppercase and used in a switch-case statement to determine which specific network's gas price fetcher function to call.

    Depending on the network value, it will execute the corresponding function that connects to the specified blockchain network (Polygon, BEP20, ERC20, BTTC, or TRON) and retrieves the current gas price.

    These gas price fetcher functions will make the necessary connections to their respective networks, retrieve the gas price data, format it, and then send it back as a response to the client.

In short, getGasPrice function is a dispatcher that redirects the request to the appropriate function that can retrieve the current gas price from the specified blockchain network.

### PR DESCRIPTION
**Pull request by kachourihassen**
Change code in get gas price best practice
We create a best practice code that :
    It takes in a request req and a response res. The request contains the parameter network, which represents the blockchain network for which you want to fetch the gas price.

    The network parameter from the request is converted to uppercase and used in a switch-case statement to determine which specific network's gas price fetcher function to call.

    Depending on the network value, it will execute the corresponding function that connects to the specified blockchain network (Polygon, BEP20, ERC20, BTTC, or TRON) and retrieves the current gas price.

    These gas price fetcher functions will make the necessary connections to their respective networks, retrieve the gas price data, format it, and then send it back as a response to the client.

In short, getGasPrice function is a dispatcher that redirects the request to the appropriate function that can retrieve the current gas price from the specified blockchain network.

### CODE REVIEW

File: controllers/wallet.controller.js [hunk 1]
```diff
@@ -172,97 +170,80 @@ exports.userBalance = async (req, res) => {
...
 
 exports.getGasPrice = async (req, res) => {
-    let network = req.params.network.toUpperCase()
-    if (network === 'TRON') {
-        let tronWeb = await webTronInstance()
+    const network = req.params.network.toUpperCase()
+
+    switch (network) {
+        case 'POLYGON':
+            return exports.gasPricePolygon(req, res)
```
Comment (hichri-louay) [inline controllers/wallet.controller.js]: I've noticed that we're using the `exports` keyword in this function, but it seems like we might not need it in this context. If this function is being used within this module and not being imported elsewhere, we can safely remove the exports keyword. This will keep the code cleaner and more readable. 
Please consider this modification.

**Review by hichri-louay — CHANGES_REQUESTED** [review]

File: routes/wallet.routes.js [hunk 2]
```diff
@@ -159,73 +157,92 @@ router.post('/userBalance', verifyAuth, walletVersionValidation ,userBalance)
...
+//  *     produces:
+//  *       - application/json
+//  *     responses:
+//  *       "200":
+//  *          description: code,<br>message:"success"
+//  *       "500":
+//  *          description: error:"error"
+//  */
+
+// router.get('/Bep20GasPrice', gasPriceBep20)
```
Comment (hichri-louay) [inline routes/wallet.routes.js]: I noticed that the router API endpoints for `gasPriceBep20`, `gasPriceBtt`, `gasPriceTrx`, `gasPricePolygon`, and `erc20GasPrice` have been removed. I understand the intention behind this, but I recommend we revert this change for now.

We're currently dependent on these endpoints on the frontend, and removing them at this stage would cause some issues. Let's plan to update the frontend first to handle these changes, and then we can safely remove these APIs without disrupting our application's functionality.

Once we've made the necessary frontend updates, we can proceed with cleaning up these APIs. This approach will ensure a smooth transition without causing any service interruptions.

**Review by hichri-louay — CHANGES_REQUESTED** [review]

File: controllers/wallet.controller.js [hunk 3]
```diff
@@ -172,97 +170,80 @@ exports.userBalance = async (req, res) => {
 }
 
 exports.getGasPrice = async (req, res) => {
```
Comment (hichri-louay) [inline controllers/wallet.controller.js:172]: I noticed that there are two functions that have the same name. This could lead to confusion and unexpected behavior, as it's unclear which function would be called at runtime.

To improve readability, I'd suggest giving each function a unique, descriptive name that reflects its purpose or the functionality it provides. 

This way, anyone reading the code can understand what each function does just by looking at its name.

File: controllers/wallet.controller.js [hunk 4]
```diff
@@ -172,97 +170,80 @@ exports.userBalance = async (req, res) => {
...
+        case 'BTTC':
+            return exports.gasPriceBtt(req, res)
+        case 'TRON':
+            return exports.gasPriceTrx(req, res)
+    }
+}
+//Best Practices
 
-        var gasPrice = await tronWeb.trx.getChainParameters()
+const getGasPrice = async (res, connect, format) => {
```
Comment (hichri-louay) [inline controllers/wallet.controller.js]: I noticed that there are two functions that have the same name. This could lead to confusion and unexpected behavior, as it's unclear which function would be called at runtime.

To improve readability, I'd suggest giving each function a unique, descriptive name that reflects its purpose or the functionality it provides. 

This way, anyone reading the code can understand what each function does just by looking at its name.

**Review by hichri-louay — CHANGES_REQUESTED** [review]

**Review by ksibisamir — APPROVED** [review]
